presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -699,6 +701,52 @@ private TableFinishNode createTemporaryTableWrite(
             SchemaTableName temporaryTableName = metadata.getTableMetadata(session, tableHandle).getTable();
             InsertHandle insertHandle = new InsertHandle(insertTableHandle, new SchemaTableName(temporaryTableName.getSchemaName(), temporaryTableName.getTableName()));
 
+            TableWriterNode tableWriter = new TableWriterNode(
+                    idAllocator.getNextId(),
+                    gatheringExchange(
+                            idAllocator.getNextId(),
+                            LOCAL,
+                            new ExchangeNode(
+                                    idAllocator.getNextId(),
+                                    REPARTITION,
+                                    REMOTE_STREAMING,
+                                    new PartitioningScheme(
+                                            Partitioning.create(partitioningHandle, partitioningVariables),
+                                            outputs,
+                                            Optional.empty(),
+                                            false,
+                                            Optional.empty()),
+                                    sources,
+                                    inputs,
+                                    Optional.empty())),
+                    insertHandle,
+                    variableAllocator.newVariable("partialrows", BIGINT),
+                    variableAllocator.newVariable("partialfragments", VARBINARY),
+                    variableAllocator.newVariable("partialtablecommitcontext", VARBINARY),
+                    outputs,
+                    outputColumnNames,
+                    Optional.of(new PartitioningScheme(
+                            Partitioning.create(partitioningHandle, partitioningVariables),
+                            outputs,
+                            Optional.empty(),
+                            false,
+                            Optional.empty())),
+                    Optional.empty());
+
+            PlanNode tableWriterMerge = tableWriter;
+            if (isTableWriterMergeOperatorEnabled(session)) {
```

Comment:
nit: Do we want to add `TableWriterMergeNode` unconditionally? -- Although the current plan doesn't hurt.

Reply:
This is more of a safety flag for rollout. I think it should be `true` by default.